plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/TrinoS3FileSystem.java

```diff
@@ -938,9 +944,36 @@ private AWSCredentialsProvider createAwsCredentialsProvider(URI uri, Configurati
                 .orElseGet(DefaultAWSCredentialsProviderChain::getInstance);
 
         if (iamRole != null) {
+            String stsEndpointOverwrite = conf.get(STS_ENDPOINT);
+            String stsRegionOverwrite = conf.get(STS_REGION);
+
+            AWSSecurityTokenServiceClientBuilder stsClientBuilder =
+                    AWSSecurityTokenServiceClientBuilder.standard()
+                    .withCredentials(provider);
+
+            if (!isNullOrEmpty(stsEndpointOverwrite) && !isNullOrEmpty(stsRegionOverwrite)) {
+                stsClientBuilder.withEndpointConfiguration(
+                    new EndpointConfiguration(stsEndpointOverwrite, stsRegionOverwrite));
+            }
+            else if (!isNullOrEmpty(stsRegionOverwrite)) {
+                stsClientBuilder.withRegion(stsRegionOverwrite);
+            }
+            else {
+                DefaultAwsRegionProviderChain regionProviderChain =
+                        new DefaultAwsRegionProviderChain();
+                try {
+                    stsClientBuilder.withRegion(regionProviderChain.getRegion());
+                }
+                catch (SdkClientException ex) {
+                    final String defaultRegion = "us-east-1";
+                    log.debug("Falling back to default AWS region " + defaultRegion);
+                    stsClientBuilder.withRegion(defaultRegion);
```

Comment:
I think we should throw here instead hard-coding failover to `us-east-1`

Reply:
Thank you for looking into the PR. Your review directly contradicts @alexjo2144 's review comment earlier on, unfortunately. I am really okay with both, I would just like to know which way to take to get this PR approved.

Reply:
For what it's worth, we default to us-east-1 when creating the `AmazonS3` client object: https://github.com/trinodb/trino/blob/master/plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/TrinoS3FileSystem.java#L885

Reply:
fair :)